openedx/core/djangoapps/programs/management/commands/backpopulate_program_credentials.py

```diff
@@ -0,0 +1,111 @@
+from collections import namedtuple
+import logging
+
+from django.core.management import BaseCommand, CommandError
+from django.db.models import Q
+from opaque_keys.edx.keys import CourseKey
+from provider.oauth2.models import Client
+
+from certificates.models import GeneratedCertificate
+from openedx.core.djangoapps.programs.models import ProgramsApiConfig
+from openedx.core.djangoapps.programs.tasks.v1.tasks import award_program_certificates
+from openedx.core.djangoapps.programs.utils import get_programs
+
+
+# TODO: Log to console, even with debug mode disabled?
+logger = logging.getLogger(__name__)
+RunMode = namedtuple('RunMode', ['course_key', 'mode_slug'])
+
+
+class Command(BaseCommand):
+    """Management command for backpopulating missing program credentials.
+
+    The command's goal is to pass a narrow subset of usernames to an idempotent
+    Celery task for further (parallelized) processing.
+    """
+    help = 'Backpopulate missing program credentials.'
+    client = None
+    run_modes = None
+    usernames = None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-c', '--commit',
+            action='store_true',
+            dest='commit',
+            default=False,
+            help='Submit tasks for processing.'
+        )
+
+    def handle(self, *args, **options):
+        programs_config = ProgramsApiConfig.current()
+        self.client = Client.objects.get(name=programs_config.OAUTH2_CLIENT_NAME)
+
+        if self.client.user is None:
+            msg = (
+                'No user is associated with the %s OAuth2 client. No tasks have been enqueued. '
+                'Associate a user with the client and try again.'
+            ).format(programs_config.OAUTH2_CLIENT_NAME)
+
+            raise CommandError(msg)
+
+        self._get_run_modes()
+
+        logger.info('Looking for users who may be eligible for a program certificate.')
+
+        self._get_usernames()
+
+        if options.get('commit'):
+            logger.info('Enqueuing tasks for %d eligible users.', len(self.usernames))
+        else:
+            logger.info(
+                'Found %d eligible users. To enqueue tasks, pass the -c or --commit flags.',
+                len(self.usernames)
+            )
+            return
+
+        succeeded, failed = 0, 0
+        for username in self.usernames:
+            try:
+                award_program_certificates.delay(username)
+            except:  # pylint: disable=bare-except
+                failed += 1
+                logger.exception('Failed to enqueue program certification task for user [%s]', username)
```

Comment:
That's what `logger.exception` does. See https://docs.python.org/2/library/logging.html#logging.Logger.exception.